pil-analyzer/src/pil_analyzer.rs

```diff
@@ -293,8 +293,10 @@ impl<T: FieldElement> PILAnalyzer<T> {
 
     fn handle_namespace(&mut self, name: SymbolPath, degree: ::powdr_ast::parsed::Expression<T>) {
         let degree = ExpressionProcessor::new(self.driver()).process_expression(degree);
+        // TODO we should maybe implement a separate evaluator that is able to run before type checking
```

Comment:
Maybe we can also postpone both namespace degree evaluation and array length evaluation: The type checker only has a concept of dynamic arrays and the actual length is checked in the condenser, which is the component that is also field-aware.

Reply:
Created an issue: https://github.com/powdr-labs/powdr/issues/1136